### PR TITLE
VZ-10244 Modify the error when the session no longer active

### PR DESCRIPTION
**Pull request by ayaranga**

### CODE REVIEW

File: pkg/auth/providers/oidc/oidc_provider.go [hunk 1]
```diff
@@ -400,6 +400,11 @@ func (o *OpenIDCProvider) getUserInfo(ctx *context.Context, config *v32.OIDCConf
 	}
 	reusedToken, err := oauth2.ReuseTokenSource(oauth2Token, oauthConfig.TokenSource(updatedContext, oauth2Token)).Token()
 	if err != nil {
+		if strings.Contains(err.Error(), "Token is not active") ||
```
Comment (lake-of-dreams) [inline pkg/auth/providers/oidc/oidc_provider.go]: should this error checking be externalized to some function where both the functions can use it for error checking?

File: pkg/auth/providers/keycloakoidc/keycloak_provider.go [hunk 2]
```diff
@@ -64,6 +64,11 @@ func (k *keyCloakOIDCProvider) newClient(config *v32.OIDCConfig, token v3.Token)
 	// get, refresh and update token
 	oauthToken, err := k.getRefreshAndUpdateToken(ctx, oauthConfig, token)
 	if err != nil {
+		if strings.Contains(err.Error(), "Token is not active") ||
+			strings.Contains(err.Error(), "Session not active") ||
+			strings.Contains(err.Error(), "invalid token") {
+			err = errors.Errorf("Session no longer active. Ignore the error")
```
Comment (lake-of-dreams) [inline pkg/auth/providers/keycloakoidc/keycloak_provider.go]: Not sure if first printing the message as an error and then saying "Ignore the error" is a correct thing to do, @kminder may comment pls?

Comment (kminder) [reply pkg/auth/providers/keycloakoidc/keycloak_provider.go]: I agree it should not say to ignore the error.  The message should tell the user what to expect.  Something like
"Session refresh will be retried."
or better
"Session refresh will be retried for N minutes."

**Review by lake-of-dreams — DISMISSED** [review]
lgtm

File: pkg/auth/util/common_util.go [hunk 3]
```diff
@@ -27,6 +29,16 @@ func ReturnHTTPError(w http.ResponseWriter, r *http.Request, httpStatus int, err
 	enc.Encode(err)
 }
 
+// ModifyErrorForInactiveSession modifies the error message for inactive token or session error
+func ModifyErrorForInactiveSession(err error) error {
```
Comment (lake-of-dreams) [inline pkg/auth/util/common_util.go]: probably should move in end of file

File: pkg/auth/providers/keycloakoidc/keycloak_provider.go (hunk 2 above)
Comment (ayaranga) [reply pkg/auth/providers/keycloakoidc/keycloak_provider.go]: @kminder it is complex to get the retry time at this point of code.  Changed it to "Session is not active, token refresh will be retried" after discussing with Ravi. Thanks.

**Review by lake-of-dreams — APPROVED** [review]

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**